python/ambassador/ir/irambassador.py

```diff
@@ -276,6 +276,33 @@ def add_mappings(self, ir: 'IR', aconf: Config):
                 mapping.referenced_by(self)
                 ir.add_mapping(aconf, mapping)
 
+        if ir.edge_stack_allowed:
+            if self.diagnostics and self.diagnostics.get("enabled", False):
```

Comment:
`False` by default? would this break existing installs?

Reply:
Hmmm. So this is mimicking the logic that we used for the old diag service - cf irambassador.py line 188 - where, basically, if you have no `diagnostics` key, then we enable it, but if you list `diagnostics` then you need to be explicit about `enabled`. We could change that...

Reply:
On Monday in #dev-general, I described this by writing
```yaml

diag_port: int           # optional, default is 8877
liveness_probe:          # optional, default is {enabled: true}
  enabled: bool            # optional, default is false
  prefix:  "path"          # optional, default is "/ambassador/v0/check_alive"
  rewrite: "path"          # optional, default is "/ambassador/v0/check_alive"
  service: "host[:port]"   # optional, default is f"127.0.0.1:{diag_port}"
readiness_probe:         # optional, default is {enabled: true}
  enabled: bool            # optional, default is false
  prefix:  "path"          # optional, default is "/ambassador/v0/check_ready"
  rewrite: "path"          # optional, default is "/ambassador/v0/check_ready"
  service: "host[:port]"   # optional, default is f"127.0.0.1:{diag_port}"
diagnostics:             # optional, default is {enabled: true}
  enabled: bool            # optional, default is false
  prefix:  "path"          # optional, default is "/ambassador/v0/"
  rewrite: "path"          # optional, default is "/ambassador/v0/"
  service: "host[:port]"   # optional, default is f"127.0.0.1:{diag_port}"
```
The `false` default here only comes in to play if the `{enabled: true}` default in `__init__()` on line 101 doesn't get applied (because other fields in `.diagnostics` are set).

Reply:
On Monday in #dev-general, I described this by writing
```yaml

diag_port: int           # optional, default is 8877
liveness_probe:          # optional, default is {enabled: true}
  enabled: bool            # optional, default is false
  prefix:  "path"          # optional, default is "/ambassador/v0/check_alive"
  rewrite: "path"          # optional, default is "/ambassador/v0/check_alive"
  service: "host[:port]"   # optional, default is f"127.0.0.1:{diag_port}"
readiness_probe:         # optional, default is {enabled: true}
  enabled: bool            # optional, default is false
  prefix:  "path"          # optional, default is "/ambassador/v0/check_ready"
  rewrite: "path"          # optional, default is "/ambassador/v0/check_ready"
  service: "host[:port]"   # optional, default is f"127.0.0.1:{diag_port}"
diagnostics:             # optional, default is {enabled: true}
  enabled: bool            # optional, default is false
  prefix:  "path"          # optional, default is "/ambassador/v0/"
  rewrite: "path"          # optional, default is "/ambassador/v0/"
  service: "host[:port]"   # optional, default is f"127.0.0.1:{diag_port}"
```
The `false` default here only comes in to play if the `{enabled: true}` default in `__init__()` on line 101 doesn't get applied (because other fields in `.diagnostics` are set).